[JIT] add builtin functions for (complex, Tensor)

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73286

Adds support in jit for op(complex, Tensor) for the following:
* add (+)
* mul (*)
* eq (==)
* ne (!=)
* sub (-)
* div (/)

These already worked in eager mode, but didn't work in jit.

Differential Revision: [D34421794](https://our.internmc.facebook.com/intern/diff/D34421794)

davidberard98 committed:
[WIP][JIT] add builtin functions for (complex, Tensor)

Adds support in jit for op(complex, Tensor) for the following:
* add (+)
* mul (*)
* eq (==)
* ne (!=)
* sub (-)
* div (/)

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] add builtin functions for (complex, Tensor)"


Adds support in jit for op(complex, Tensor) for the following:
* add (+)
* mul (*)
* eq (==)
* ne (!=)
* sub (-)
* div (/)

These already worked in eager mode, but didn't work in jit.

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34421794).

eellison commented on torch/csrc/jit/frontend/builtin_functions.cpp:
```diff
@@ -18,6 +18,14 @@ def ne(a : ${Scalar}, b : Tensor) -> Tensor:
+  return torch.reciprocal(b) * a
+)SCRIPT");
+
+auto scalar_comparator_operators_source = at::jit::CodeTemplate(
```
nit: `scalar_comparator_operators_source` -> `scalar_operators_complex_nyi_source` maybe ?

eellison reviewed (approved):
Nice!!

davidberard98 committed:
Update on "[JIT] add builtin functions for (complex, Tensor)"


Adds support in jit for op(complex, Tensor) for the following:
* add (+)
* mul (*)
* eq (==)
* ne (!=)
* sub (-)
* div (/)

These already worked in eager mode, but didn't work in jit.

Differential Revision: [D34421794](https://our.internmc.facebook.com/intern/diff/D34421794)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34421794).
